flake.nix

```diff
@@ -6,29 +6,80 @@
 
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    flake-root.url = "github:srid/flake-root";
+    treefmt-nix = {
+      url = "github:numtide/treefmt-nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    # For preserving compatibility with non-Flake users
+    flake-compat = {
+      url = "github:nix-community/flake-compat";
+      flake = false;
+    };
+    nix-fast-build = {
+      url = "github:Mic92/nix-fast-build";
+      # re-use some existing inputs
+      inputs = {
+        flake-parts.follows = "flake-parts";
+        treefmt-nix.follows = "treefmt-nix";
+      };
+    };
+    ci-public = {
+      url = "github:tiiuae/ci-public";
+      flake = false;
+    };
+    sbomnix = {
+      url = "github:brianmcgee/sbomnix/fix/repolocy_cli-reference";
+      inputs = {
+        # reduce duplicate inputs
+        nixpkgs.follows = "nixpkgs";
+        flake-root.follows = "flake-root";
+        flake-parts.follows = "flake-parts";
+        treefmt-nix.follows = "treefmt-nix";
+        nix-fast-build.follows = "nix-fast-build";
+      };
+    };
   };
 
-  outputs = {
-    self,
-    nixpkgs,
-  }: let
-    pkgs = import nixpkgs {system = "x86_64-linux";};
-    ghafscan = import ./default.nix {pkgs = pkgs;};
-    ghafscan-shell = import ./shell.nix {pkgs = pkgs;};
-  in rec {
-    # nix package
-    packages.x86_64-linux = {
-      inherit ghafscan;
-      default = ghafscan;
-    };
+  outputs = inputs @ {flake-parts, ...}:
+    flake-parts.lib.mkFlake
+    {
+      inherit inputs;
+    } {
+      systems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
 
-    # nix run .#ghafscan
-    apps.x86_64-linux.ghafscan = {
-      type = "app";
-      program = "${self.packages.x86_64-linux.ghafscan}/bin/ghafscan";
+      imports = [
+        ./nix
+      ];
     };
 
-    # nix develop
-    devShells.x86_64-linux.default = ghafscan-shell;
-  };
+  #  outputs = {
+  #    self,
+  #    nixpkgs,
+  #  }: let
+  #    pkgs = import nixpkgs {system = "x86_64-linux";};
+  #    ghafscan = import ./default.nix {pkgs = pkgs;};
+  #    ghafscan-shell = import ./shell.nix {pkgs = pkgs;};
+  #  in rec {
+  #    # nix package
+  #    packages.x86_64-linux = {
+  #      inherit ghafscan;
+  #      default = ghafscan;
+  #    };
+  #
+  #    # nix run .#ghafscan
+  #    apps.x86_64-linux.ghafscan = {
+  #      type = "app";
+  #      program = "${self.packages.x86_64-linux.ghafscan}/bin/ghafscan";
+  #    };
+  #
+  #    # nix develop
+  #    devShells.x86_64-linux.default = ghafscan-shell;
+  #  };
```

Comment:
@brianmcgee : did you intentionally leave this part here?

Reply:
Nope :)